### PR TITLE
dnn(tflite): add 3rdparty flatbuffers with pre-generated schema

### PR DESCRIPTION
**Pull request by alalek**
relates #23161
https://github.com/google/flatbuffers/releases/tag/v23.1.21 (Apache 2.0)

TODO:

- [x] fix build issues (32-bit and static libraries)
- [x] fix tests
- [x] readme and licensing information

<cut/>

```
force_builders=Custom,Custom Win,Linux AVX2,Linux OpenCL,linux,windows,docs,Win32,linux32
build_image:Custom=ubuntu:16.04
build_shared:Custom=OFF
build_examples:Custom=OFF
build_shared:Custom Win=OFF
Xbuild_shared:Linux AVX2=OFF
```

### CODE REVIEW

**Comment by dkurt** [comment]
BTW, is there limitations why OpenCV does not support submodules instead?

**Comment by alalek** [comment]
We don't use submodules in the main OpenCV repositories.

File: modules/dnn/CMakeLists.txt [hunk 1]
```diff
@@ -292,8 +299,9 @@ if(TARGET ocv.3rdparty.cann AND OPENCV_TEST_DNN_CANN)
   endif()
 endif()
 
-if(HAVE_FLATBUFFERS)
+ocv_option(OPENCV_TEST_DNN_TFLITE "Build test with TFLite" (TARGET ocv.3rdparty.flatbuffers))
```
Comment (dkurt) [inline modules/dnn/CMakeLists.txt:302]: Do we really need this option?

**Review by mshabunin — APPROVED** [review]
Overall looks good to me.

File: cmake/OpenCVDetectFlatbuffers.cmake [hunk 2]
```diff
@@ -0,0 +1,19 @@
+if(WITH_FLATBUFFERS)
+  set(HAVE_FLATBUFFERS 1)
+  set(flatbuffers_VERSION "23.1.21")
+  ocv_install_3rdparty_licenses(flatbuffers "${OpenCV_SOURCE_DIR}/3rdparty/flatbuffers/LICENSE.txt")
+  ocv_add_external_target(flatbuffers "${OpenCV_SOURCE_DIR}/3rdparty/flatbuffers/include" "" "HAVE_FLATBUFFERS=1")
+  set(CUSTOM_STATUS_flatbuffers "    Flatbuffers:" "builtin/3rdparty (${flatbuffers_VERSION})")
```
[the remark this answers is not in the document]
Comment (alalek) [reply cmake/OpenCVDetectFlatbuffers.cmake:6]: `flatbuffers` is integrated as headers only (there is no compilation of the library). IMHO, `build` may confuse here.

File: modules/dnn/CMakeLists.txt (hunk 1 above)
Comment (alalek) [reply modules/dnn/CMakeLists.txt]: Thanks! Updated handing of this option in tests. It should be more clear now (flatbuffers -> tflite)

**Pull request merged**